src/collective.jl

```diff
@@ -839,3 +839,150 @@ Exscan(sendbuf::AbstractArray, op, comm::Comm) =
     Exscan!(sendbuf, similar(sendbuf), op, comm)
 Exscan(object::T, op, comm::Comm) where {T} =
     Exscan!(Ref(object), Ref{T}(), op, comm)[]
+
+"""
+    Neighbor_alltoall!(sendbuf::UBuffer, recvbuf::UBuffer, comm::Comm)
+
+Perform an all-to-all communication along the directed edges of the graph with fixed size messages.
+
+See also [`MPI.Alltoall!`](@ref).
+
+# External links
+$(_doc_external("MPI_Neighbor_alltoall"))
+"""
+function Neighbor_alltoall!(sendbuf::UBuffer, recvbuf::UBuffer, graph_comm::Comm)
+    # int MPI_Neighbor_alltoall(const void *sendbuf, int sendcount, MPI_Datatype sendtype, void *recvbuf,
+    #       int recvcount, MPI_Datatype recvtype, MPI_Comm graph_comm)
+    try
+        @mpichk ccall((:MPI_Neighbor_alltoall, libmpi), Cint,
+                    (MPIPtr, Cint, MPI_Datatype, MPIPtr, Cint, MPI_Datatype, MPI_Comm),
+                    sendbuf.data, sendbuf.count, sendbuf.datatype,
+                    recvbuf.data, recvbuf.count, recvbuf.datatype,
+                    graph_comm)
+    catch e
+        miv_ver = v"3.0"
+        if contains(e.msg, "could not load symbol") && contains(e.msg, "MPI_Neighbor_alltoall") && MPI.Get_version() < miv_ver
+            throw(MPI.FeatureLevelError("MPI_Neighbor_alltoall", miv_ver))
+        end
+        rethrow(e)
+    end
```

Comment:
I'm not sure this is the best way to do this. @giordano @vchuravy is there a way we could check without the `try`/`catch`, that would have minimal overhead?

Reply:
To check whether a symbol is available? [`dlsym(...; throw_error=false)`](https://docs.julialang.org/en/v1/stdlib/Libdl/#Base.Libc.Libdl.dlsym)

Reply:
Is the overhead for the MPI call really this small compared to the exception handling?

Reply:
Not sure, just that this seems a bit messy. We could add the logic into `@mpichk`?

Reply:
Is this what you had in mind?

Reply:
I don't think Simon wanted to have a try/catch block in _all_ ccalls, especially since he wanted to remove the one you put here to start with 🙂

Reply:
Okay, thanks. I removed the try-catch blocks in favor of working on dlsym in the latest commit. However, I really have trouble understanding why moving the logic into the macro seem to cause the windows tests to crash sometimes.